### ISSUE TITLE
Content restriction notices not shown on a "redirected" sales page.

### ISSUE THREAD

**Issue by eri-trabiccolo** [gocodebox/lifterlms#983]
### Reproduction Steps
(reported by a user who also provided a video)
+ set a course sales page redirection to a whatever wp page.
+ as visitor visit a lesson, which requires the above course enrollment, for which you have a direct link.
+ you get redirected to the course page, and then to the sales page you set in the first step.
**(first issue)**
+ repeat the second step above, say 2 times.
+ now visit a course page (that has no sales page redirection, of course), or the checkout page, whatever "page" we know that print notices, let's say we navigate to the account login page.
**(second issue)**

### Expected Behavior

**(first issue)**
+ I would expect to still see the restriction notice anyways.
But we could discuss on whether this actually makes sense.

**(second issue)**
+ I would expect nothing special :), e.g. I would expect to see the account login page normally.

### Actual Behavior

**(first issue)**
+ notices do not show up on the redirected page.

**(second issue)**
+ Two (equal to the number of the lesson access attempts) restriction notices are shown.

### Considerations 

This is due to the fact that those notices are not printed throughout the site but only in certain (llms) templates, hence while not shown on a "redirected sales page" if you navigate to a certain (llms) page like the login one (notices printed before the login form).

I'm not sure what would be the best solution here.
Does it make sense showing that kind of notice on a "redirected sales page"?
If so, maybe we can do something of the sort:
1) Pass a GET parameter here (`add_query_arg()`):
https://github.com/gocodebox/lifterlms/blob/3.36.4/includes/class.llms.template.loader.php#L97

2) Check the GET parameter here: 
https://github.com/gocodebox/lifterlms/blob/3.36.4/includes/class.llms.template.loader.php#L396
and if true, fire the `llms_print_notices` at `loop_start`

Otherwise I don't think it would make any sense showing these notices in other pages, maybe after you've navigated through 10 different pages on the website and you land on a "llms page"..., and we can just clear the notices (`llms_clear_notices()`)
here:
https://github.com/gocodebox/lifterlms/blob/3.36.4/includes/class.llms.template.loader.php#L95

### Error Messages / Logs

n/a

### System Information
LifterLMS 3.36.4
WordPress 5.2.4

### Browser, Device, and Operating System Information

n/a

**Comment by thomasplevy** [comment]
There's a related issue to consider when talking about whether or not the sales page should display the redirect notice, see #628 

My comment on this issue has what I think it a pretty good solution to that issue: https://github.com/gocodebox/lifterlms/issues/628#issuecomment-413241932

However, displaying the notices on a regular page becomes fairly problematic as we don't actually have any way to ensure that we'll have a standard hook that we can use to output the notice(s).

Outputting an action in the template loader function is quite problematic... some templates are parts, but some templates are the full html document. If we output it at the beginning of our template loader we'll risk outputting it before the opening `<html>` tag of the document.

Maybe we can attach this to `the_content` and output the notice there. It's not perfect but it'll ensure that we don't break the DOM. 

The multiple notices displayed later is something of a bug, for sure... I think it should maybe be tracked separately. If you visit lesson 10 and all the lessons have prerequisites it'll redirect you 9 times back to lesson one adding a notice each time. We should probably be more intelligent in the restriction redirect logic so that you only get redirected once to the accessible lesson (and only see one notice).

I'm not sure how to go about fixing it with a sales page redirect though. Maybe we can check where they're being redirect to and then determine whether or not to output the notice...

However, if we can tackle displaying it properly on the sales page then we actually fix some of the issue with multiple notices (I think).

**Comment by thomasplevy** [comment]
@eri-trabiccolo was this reported in HelpScout or is there a WordPress.org forum post about this?

**Comment by eri-trabiccolo** [comment]
We talked about maybe use a pure PHP session for this purpose.
I've explored this a little bit and, although, it's of very simple implementation I don't know :D ... I don't want to open a session (hence write a cookie too) on whatever page (to check if the flag is set) for what, anyways, is not something that will happen very often.
You know, we also should make sure to open the session before anything is print out, and that would be fine... But there can always be a 3rd party code that prints something before `wp` without throwing any particular error but a notice saying:
`PHP Warning:  session_start(): Cannot start session when headers already sent ...` pointing to the line of code where we start the session.

I don't want problems on any sort for this improvement.
I'm going to go with the $_GET way...
